tests/test_diffraction_StaticStructureFactorDirect.py

```diff
@@ -0,0 +1,248 @@
+        sf.compute(system)
+        assert np.isclose(sf.S_k[0], N)
+
+    def test_accumulation(self):
+        # This test ensures that accumulation and resetting works as expected.
+        # See notes on test_S_0_is_N.
+        L = 10
+        N = 1000
+        sf = freud.diffraction.StaticStructureFactorDirect(bins=100, k_max=10)
+        # Ensure that accumulation averages correctly over different numbers of
+        # points. We test N points, N*2 points, and N*3 points. On average, the
+        # number of points is N * 2.
+        for i in range(1, 4):
+            box, points = freud.data.make_random_system(L, N * i)
+            sf.compute((box, points), reset=False)
+        assert np.isclose(sf.S_k[0], N * 2)
+        box, points = freud.data.make_random_system(L, N * 2)
+        sf.compute((box, points), reset=True)
+        assert np.isclose(sf.S_k[0], N * 2)
+
+    def test_k_min(self):
+        L = 10
+        N = 1000
+        box, points = freud.data.make_random_system(L, N)
+        system = freud.AABBQuery.from_system((box, points))
+        sf1 = freud.diffraction.StaticStructureFactorDirect(bins=100, k_max=10)
+        sf1.compute(system)
+        sf2 = freud.diffraction.StaticStructureFactorDirect(bins=50, k_max=10, k_min=5)
+        sf2.compute(system)
+        npt.assert_allclose(sf1.bin_centers[50:], sf2.bin_centers, rtol=1e-6, atol=1e-6)
+        npt.assert_allclose(sf1.bin_edges[50:], sf2.bin_edges, rtol=1e-6, atol=1e-6)
+        npt.assert_allclose(sf1.S_k[50:], sf2.S_k, rtol=1e-6, atol=1e-6)
+        with pytest.raises(ValueError):
+            freud.diffraction.StaticStructureFactorDirect(bins=100, k_max=10, k_min=-1)
+
+    def test_partial_structure_factor_arguments(self):
+        sf = freud.diffraction.StaticStructureFactorDirect(bins=1000, k_max=100)
+        box, positions = freud.data.UnitCell.fcc().generate_system(4)
+        # Require N_total if and only if query_points are provided
+        with pytest.raises(ValueError):
+            sf.compute((box, positions), query_points=positions)
+        with pytest.raises(ValueError):
+            sf.compute((box, positions), N_total=len(positions))
+
+    def test_partial_structure_factor_symmetry(self):
+        """Compute a partial structure factor and ensure it is symmetric under
+        type exchange."""
+        L = 10
+        N = 1000
+        box, points = freud.data.make_random_system(L, N)
+        system = freud.AABBQuery.from_system((box, points))
+        A_points = system.points[: N // 3]
+        B_points = system.points[N // 3 :]
+        sf = freud.diffraction.StaticStructureFactorDirect(bins=100, k_max=10)
+        sf.compute((system.box, B_points), query_points=A_points, N_total=N)
+        S_AB = sf.S_k
+        sf.compute((system.box, A_points), query_points=B_points, N_total=N)
+        S_BA = sf.S_k
+        npt.assert_allclose(S_AB, S_BA, rtol=1e-6, atol=1e-6)
```

Comment:
Tests are failing [here](https://app.circleci.com/pipelines/github/glotzerlab/freud/3165/workflows/5359a0d2-ac86-4dd3-80cf-b45dd7ee63b0/jobs/12387) with:
> Max absolute difference: 1.4603138e-06
> Max relative difference: 0.00017976

We should increase this tolerance.
```suggestion
        npt.assert_allclose(S_AB, S_BA, rtol=1e-5, atol=1e-5)
```